space-concordia-robotics/robotics-prototype
# Warning 342

# Assignee Section

## Description
Add emphasis message when battery voltage or temperature is nearing danger zone (yellow) and add warning sound when battery voltage is in danger zone.

- Added js library notify.js
- Added warning pop op message for battery voltage and temperature
- Added warning sound file warning.wav
- Added warningSound function with the responsibility of triggering the warning sound.
- Replaced call to errorSound function with warningSound when battery voltage is in danger zone

### Steps for Testing

- Open Gui
- Call rostopic command for voltage when nearing danger zone (12.5 < voltage< 13.5):rostopic pub -r 0.2 /battery_voltage mcu_control/Voltage 13
The warning pop up message for battery voltage should appear on the top right of the GUI below the battery temperature.
- Call rostopic command for battery temperatures when nearing danger zone (0<= temperature <10 || 75< temperature <=85) : rostopic pub -r 0.2 mcu_control/ThermistorTemps "therm1: 34.0
therm2: 40.0
therm3: 2.0"
The warning pop up for battery temperature should appear on the top right below the leftest battery temperature
- Call rostopic for voltage when in danger zone (voltage<12.5 || voltage> 16.8) : rostopic pub -r 0.2 /battery_voltage std_msgs/Float32 17
Warning Sound should be heard only once alongside an exit pop indicating to either charge or discharge the battery voltage.


closes #342

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
